### PR TITLE
BM40 keymap for Skewwhiffy, revamp

### PR DESCRIPTION
**Pull request by skewwhiffy**
## Description

* New Colemak-based layout for BM40 keyboard

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/kprepublic/bm40hsrgb/keymaps/skewwhiffy/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+!keyboard-layout.png
```
Comment (waffle87) [inline keyboards/kprepublic/bm40hsrgb/keymaps/skewwhiffy/.gitignore:1]: Did you intend for this image to be included or not? If yes, it needs to be rehosted on Imgur and displayed in `readme.md` using that link.

File: keyboards/kprepublic/bm40hsrgb/keymaps/skewwhiffy/keymap.c [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+    KC_TAB,    KC_LEFT, KC_DOWN, KC_RGHT, _______, RGB_SAI, _______, _______, KC_INS,  KC_HOME, KC_PGUP, _TERM,   \
+    _S_TAB,    _A_LEFT, _______, __NAL,   _______, RGB_VAI, QK_BOOT, _______, KC_DEL,  KC_END,  KC_PGDN, _______, \
+    _______,   _______, _______, _______, _______,       RGB_MOD,    _______, _______, _______, _______, _______  \
+  ),
+  [nar] = LAYOUT_planck_mit( \
+    _VIM_ESC,  _C_LEFT, KC_UP,   _C_RGHT, _______, RGB_HUI, RGB_TOG, _______, KC_PSCR, KC_SCRL, KC_PAUS, KC_ESC,  \
+    KC_TAB,    KC_LEFT, KC_DOWN, KC_RGHT, _______, RGB_SAI, _______, _______, KC_INS,  KC_HOME, KC_PGUP, _TERM,   \
+    _S_TAB,    _A_LEFT, _______, _A_RGHT, _______, RGB_VAI, QK_BOOT, _______, __NAR,   KC_END,  KC_PGDN, _______, \
+    _______,   _______, _______, _______, _______,      RGB_MOD,     _______, _______, _______, _______, _______ \
+  ),
```
Comment (waffle87) [inline keyboards/kprepublic/bm40hsrgb/keymaps/skewwhiffy/keymap.c:150]:
```suggestion
  [cm] = LAYOUT_planck_mit(
    UK_Q,    UK_W,    UK_F,    UK_P,    UK_G, _______, _______, UK_J,    UK_L,    UK_U,    UK_Y,    UK_SCLN,
    UK_A,    UK_R,    UK_S,    UK_T,    UK_D, _______, _______, UK_H,    UK_N,    UK_E,    UK_I,    UK_O,
    _Z_SFT,  _X_NB,   _C_SY,   _V_NAL,  UK_B, KC_DEL,  KC_DEL,  UK_K,    _M_NAR,  _COM_SY, _DOT_NB, _SLSH,
    KC_LSFT, KC_LCTL, KC_LALT, KC_LGUI, KC_SPC,     KC_ENT,     KC_BSPC, KC_RGUI, KC_RALT, KC_RCTL, KC_RSFT
  ),

/* Numbers _NB
 * .-----------------------------------------------------------------------------------------------------------.
 * |        | F7     | F8     | F9     | F10    |        |        |        | 7      | 8      | 9      |        |
 * |-----------------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------|
 * |        | F4     | F5     | F6     | F11    |        |        |        | 4      | 5      | 6      |        |
 * |-----------------+--------+--------+--------+--------+--------+--------+--------------------------+--------|
 * |        | F1     | F2     | F3     | F12    |        |        | 0      | 1      | 2      | 3      | .      |
 * |-----------------+--------+--------+--------+--------+--------+--------+-----------------+--------+--------|
 * |        |        |        |        |        |        |        |        |        |        |        |        |
 * '-----------------------------------------------------------------------------------------------------------'
 */
  [nbl] = LAYOUT_planck_mit(
    _______, KC_F7,   KC_F8,   KC_F9,   KC_F10,  _______, _______, _______, KC_7,    KC_8,    KC_9,    _______,
    _______, KC_F4,   KC_F5,   KC_F6,   KC_F11,  _______, _______, _______, KC_4,    KC_5,    KC_6,    _______,
    _______, __NBL,   KC_F2,   KC_F3,   KC_F12,  _______, _______, KC_0,    KC_1,    KC_2,    KC_3,    KC_DOT,
    _______, _______, _______, _______, _______,      _______,     _______, _______, _______, _______, _______
  ),
  [nbr] = LAYOUT_planck_mit( \
    _______, KC_F7,   KC_F8,   KC_F9,   KC_F10,  _______, _______, _______, KC_7,    KC_8,    KC_9,    _______,
    _______, KC_F4,   KC_F5,   KC_F6,   KC_F11,  _______, _______, _______, KC_4,    KC_5,    KC_6,    _______,
    _______, KC_F1,   KC_F2,   KC_F3,   KC_F12,  _______, _______, KC_0,    KC_1,    KC_2,    __NBR,   KC_DOT,
    _______, _______, _______, _______, _______,      _______,     _______, _______, _______, _______, _______
  ),

/* Symbols _SY
 * .-----------------------------------------------------------------------------------------------------------.
 * | !      | £      | _      | -      | ~      |        |        | \      | {      | }      | /      | #      |
 * |--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------|
 * | $      | %      | +      | =      |        |        |        | "      | (      | )      | '      | @      |
 * |--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------|
 * | ^      | &      | *      | |      |        |        |        | <      | [      | ]      | >      | `      |
 * |--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------|
 * |        |        |        |        |        |        |        |        |        |        |        |        |
 * '-----------------------------------------------------------------------------------------------------------'
 */
  [syl] = LAYOUT_planck_mit(
    UK_EXLM, UK_PND,  UK_UNDS, UK_MINS, UK_TILD, _______, _______, UK_BSLS, UK_LCBR, UK_RCBR, UK_SLSH, UK_HASH,
    UK_DLR,  UK_PERC, UK_PLUS, UK_EQL,  _______, _______, _______, UK_DQUO, UK_LPRN, UK_RPRN, UK_QUOT, UK_AT,
    UK_CIRC, UK_AMPR, __SYL,   UK_PIPE, _______, _______, _______, UK_LABK, UK_LBRC, UK_RBRC, UK_RABK, UK_GRV,
    _______, _______, _______, _______, _______,      _______,     _______, _______, _______, _______, _______
  ),
  [syr] = LAYOUT_planck_mit(
    UK_EXLM, UK_PND,  UK_UNDS, UK_MINS, UK_TILD, _______, _______, UK_BSLS, UK_LCBR, UK_RCBR, UK_SLSH, UK_HASH,
    UK_DLR,  UK_PERC, UK_PLUS, UK_EQL,  _______, _______, _______, UK_DQUO, UK_LPRN, UK_RPRN, UK_QUOT, UK_AT,
    UK_CIRC, UK_AMPR, UK_ASTR, UK_PIPE, _______, _______, _______, UK_LABK, UK_LBRC, __SYR,   UK_RABK, UK_GRV,
    _______, _______, _______, _______, _______,      _______,     _______, _______, _______, _______, _______
 ),

/*
 * Navigation
 * .----------------------------------------------------------------------------------------------------------------.
 * | Ctrl [ | Ctrl L | Up     | Ctrl R |        | RGB HUE | RGB ON/OFF |        | PtSn   | ScLk   | Pause  | ESC    |
 * +--------+--------+--------+--------+--------+---------+------------+--------+--------+--------+--------+--------|
 * | Tab    | Left   | Down   | Right  |        | RGB SAT |            |        | Insert | Home   | PgUp   | Term   |
 * +--------+--------+--------+--------+--------+---------+------------+--------+--------+--------+--------+--------|
 * | ShTab  | Alt L  |        | Alt R  |        | RGB VAL | RESET      |        | Delete | End    | PgDn   |        |
 * +--------+--------+--------+--------+--------+---------+------------+--------+--------+--------+--------+--------|
 * |        |        |        |        |        |     RGB MODE         |        |        |        |        |        |
 * '----------------------------------------------------------------------------------------------------------------'
 */
  [nal] = LAYOUT_planck_mit(
    _VIM_ESC,  _C_LEFT, KC_UP,   _C_RGHT, _______, RGB_HUI, RGB_TOG, _______, KC_PSCR, KC_SCRL, KC_PAUS, KC_ESC,
    KC_TAB,    KC_LEFT, KC_DOWN, KC_RGHT, _______, RGB_SAI, _______, _______, KC_INS,  KC_HOME, KC_PGUP, _TERM,
    _S_TAB,    _A_LEFT, _______, __NAL,   _______, RGB_VAI, QK_BOOT, _______, KC_DEL,  KC_END,  KC_PGDN, _______,
    _______,   _______, _______, _______, _______,       RGB_MOD,    _______, _______, _______, _______, _______
  ),
  [nar] = LAYOUT_planck_mit(
    _VIM_ESC,  _C_LEFT, KC_UP,   _C_RGHT, _______, RGB_HUI, RGB_TOG, _______, KC_PSCR, KC_SCRL, KC_PAUS, KC_ESC,
    KC_TAB,    KC_LEFT, KC_DOWN, KC_RGHT, _______, RGB_SAI, _______, _______, KC_INS,  KC_HOME, KC_PGUP, _TERM,
    _S_TAB,    _A_LEFT, _______, _A_RGHT, _______, RGB_VAI, QK_BOOT, _______, __NAR,   KC_END,  KC_PGDN, _______,
    _______,   _______, _______, _______, _______,      RGB_MOD,     _______, _______, _______, _______, _______
  )
```

File: keyboards/kprepublic/bm40hsrgb/keymaps/skewwhiffy/readme.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
+# Skewwhiffy keymap
+
+## Overview
+
+A small-form ergonomic Colemak-based multi-layered layout, optimized for software engineering. Programmed to emulate a standard UK ISO keyboard.
+
+Keyboard layout can be viewed [here](http://www.keyboard-layout-editor.com/#/gists/470f01e934c27d41491dea9ef5c6fde6)
+
+![Keyboard layout](./keyboard-layout.png)
```
Comment (waffle87) [inline keyboards/kprepublic/bm40hsrgb/keymaps/skewwhiffy/readme.md:9]: Above comment wrt. to layout image.

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/kprepublic/bm40hsrgb/keymaps/skewwhiffy/.gitignore (hunk 1 above)
Comment (skewwhiffy) [reply keyboards/kprepublic/bm40hsrgb/keymaps/skewwhiffy/.gitignore:1]: Unfortunately, I cannot register on Imgur: it's blocked for new registrations in the UK atm.

Using postimage, hope that's okay.

File: keyboards/kprepublic/bm40hsrgb/keymaps/skewwhiffy/keymap.c (hunk 2 above)
Comment (skewwhiffy) [reply keyboards/kprepublic/bm40hsrgb/keymaps/skewwhiffy/keymap.c]: Done.

File: keyboards/kprepublic/bm40hsrgb/keymaps/skewwhiffy/readme.md (hunk 3 above)
Comment (skewwhiffy) [reply keyboards/kprepublic/bm40hsrgb/keymaps/skewwhiffy/readme.md]: I've also squashed my PR so that the image is no longer in git at all.

**Review by waffle87 — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**